### ISSUE TITLE
dollyToCursor + dollyTo

### ISSUE THREAD

**Issue by Muffles** [yomotsu/camera-controls#267]
**Is your feature request related to a problem? Please describe.**
I'm working on a project where dollyToCursor is invaluable, however, quite often I need to programmatically control and listen to the zoom/distance changes. This means using for example dollyTo(). Using this, as it seems, doesn't allow me to combine my need to "dolly in/out" with a specific controlled speed, while at the same time do it towards a cursor point.

**Describe the solution you'd like**
Ideally I would like to be able to combine these two functionalities in a "dollyToCursor(distance)".
Maybe there is another way to combine both these aspects, I just haven't seemed to find it.
I believe exposing `_dollyInternal` in some other format, would suffice (I'm using it atm).

**Describe alternatives you've considered**
I've tried many combinations of settings, and even to combine dollyTo and moveTo in order to fake a move to cursor, but failed so far.

**Comment by yomotsu** [comment]
Thanks for using the camera-controls.
Can the functionality be done with `moveTo()` and dolly To()`?

Or here is a workaround:
The internal mouse wheel dolly can be emulated with a tween library.
```js

const dollyTo = 1;
const dollyToX = 1; // - 1 (left) to 1 (right)
const dollyToY = 0; // - 1 (bottom) to 1 (top)

dollyToPoint( dollyTo, dollyToX, dollyToY );

// ---

function dollyToPoint( dollyTo, dollyToX, dollyToY ) {

  cameraControls.dollyToCursor = true;
  cameraControls._getClientRect( cameraControls._elementRect );

  const currentDistance = cameraControls.distance;
  const progress = { value: 0 };
  gsap.fromTo(
    progress,
    {
      value: currentDistance,
    },
    {
      value: dollyTo,
      duration: 1,
      paused: false,
      onUpdateParams: [ progress ],
      onUpdate( { value } ) {

        const prevRadius = cameraControls._sphericalEnd.radius;
        cameraControls.dollyTo( value );
        cameraControls._dollyControlAmount += cameraControls._sphericalEnd.radius - prevRadius;
        cameraControls._dollyControlCoord.set( dollyToX, dollyToY );

      },
    }
  );

}
```

**Comment by Muffles** [comment]
Hi there, appreciate the follow up!

I ended up implementing a simpler approach by removing the wheel actions, and manually listening to zoom events.
Then I'm able to tap into the library's `_dollyInternal`. I had to reverse-engineer the calculations happening inside of it in order to provide the correct ratio based on the distance I want to travel to:

```
    // ...
    
    const _distance = 1000; // or any other distance
    const _dollyscale = _distance / cameracontrols._sphericalEnd.radius;
    const _delta = Math.log(_dollyscale)/Math.log(0.95);

    cameracontrols._dollyInternal(_delta * -1, x, y);
    
    // ...
```

As far as I can test, I'm able to navigate towards a desired distance while directed to the cursor.
